### ISSUE TITLE
Fog modes in ViewportEvent.RenderFog do not work correctly

### ISSUE THREAD

**Issue by FlashFyreMods** [MinecraftForge/MinecraftForge#8907]
**Minecraft Version:** 1.19

**Forge Version:** 41.0.110

**Logs:** Not relevant

**Steps to Reproduce:**
 1. Add an event listener to ViewportEvent.RenderFog
 2. Check if the fog mode is FOG_TERRAIN and use RenderSystem#setShaderFogColour. I've set it to a shade of lime with rgb values of 0.35F, 0.9F, 0.3F.
 3. Load up the game and enter the overworld. It should look correct thus far, with green fog once the chunks have loaded in.
 4. Change the mode in the if statement to check for FogMode.FOG_SKY, the sky fog (the fog rendered behind blocks) will NOT have changed colour to green - no green is visible at all.
 5. Don't change it back to the terrain mode yet. Use the command "/execute in minecraft:the_nether run tp ~ ~ ~" to teleport into the nether. This time the fog is green, but for the terrain fog, not the sky fog, which shouldn't happen. (I'm not actually sure if the nether even has any sky fog, but even so it shouldn't be changing the terrain fog when the mode isn't set to terrain).
 6. Change the mode to check for back to FOG_TERRAIN, also perhaps change the rgb values so you know it has changed. Produces same result as FOG_SKY in the nether. 

Checking for FOG_TERRAIN in overworld and setting as green
![image](https://user-images.githubusercontent.com/33395370/181090460-7789bc3b-62ed-464c-afec-2c9c9c30396a.png)
Checking for FOG_SKY in overworld and setting as green
![image](https://user-images.githubusercontent.com/33395370/181090564-60a7ee96-b629-45c3-b8a2-051ca2dae021.png)
Checking for FOG_SKY or FOG_TERRAIN in the nether and setting as green
![image](https://user-images.githubusercontent.com/33395370/181091077-a78cecbd-ef95-44f0-9fa5-0a90206c2765.png)

**Description of issue:**
Likely related to #8825.
In the overworld, changing the fog colour using RenderSystem#setShaderFogColour in a ViewportEvent.RenderFog event has no affect on the fog using the FOG_SKY fogmode. The FOG_TERRAIN fogmode works as intended. In the nether, no matter which fogmode is used, it always changes the terrain fog, not the sky fog even if you check if the fogmode is FOG_SKY.
Tldr: sky fog mode does not work as intended with different results based on dimension.

**Comment by sciwhiz12** [comment]
From what I can tell from testing and looking at the code, it seems to me that the fog color is shared by the sky and terrain fog. Setting the color of the sky fog in the overworld seems to do nothing if it is currently midday. However, at other points such as during the night, changing the color of the sky fog also modifies the color of the terrain fog.

`ViewportEvent.RenderFog` seems to be misnamed a bit, since it doesn't fire when the fog is rendered, but rather when the setup for the fog, terrain or sky, is done. The actual rendering is done at some later point (which I am unable to pinpoint, as I can't fully make out how rendering is done).

Also, I believe the Nether does have sky fog, but because the color is shared between the terrain and sky fog, one cannot modify only the sky fog color or only the terrain fog color.

**Comment by eerussianguy** [comment]
Do we not have a ComputeFogColor event for computing the fog color? Why are you using the render fog event for this? Either way the correct RenderSystem call is RenderSystem#clearColor

**Comment by Jonathing** [comment]
I forgot to mention this earlier, but I think this issue was created based on a misunderstanding that both @FlashFyreMods and I (we work on the same project) had about how fog works. Basically, I agree with what @sciwhiz12 said about the method being a tad misnamed, since it is mainly about the fog setup and the distances it will use. I have since made changes in my own project to reflect that using `RenderFog` and `ComputeFogColor` instead of manually making calls to `RenderSystem.setShaderFogColour()`.
